my-project/src/Controller/ProductCategoryCSVController.php

```diff
@@ -71,4 +72,50 @@ public function exportCSV(ProductCategory $productCategory, SerializerService $s
             $dataJson
             ,200);
     }
+
+    /**
+     * @Route("/", name="product_category_import_csv_api", methods="POST")
+     */
+    public function importFromCSV(Request $request, SerializerInterface $serializer, ValidatorInterface $validator)
+    {
+        $productCategoriesArray = $serializer->decode(file_get_contents($request->get('csvFile')), 'csv');
+        $em = $this->getDoctrine()->getManager();
+        $tmpProductsCategories = [];
+        foreach ($productCategoriesArray as $productCategory) {
+            $tmpProductCategory = new ProductCategory();
+            $tmpProductCategory->setName($productCategory['name']);
+            $tmpProductCategory->setDescription($productCategory['description']);
+            $tmpProductCategory->setMainImage($productCategory['mainImage']);
+            $tmpProductCategory->setDateOfCreation(new \DateTime($productCategory['dateOfCreation']));
+            $tmpProductCategory->setDateOfLastModification(new \DateTime($productCategory['dateOfLastModification']));
+
+            $imgPath = $tmpProductCategory->getMainImage();
+            $tmpProductCategory->setMainImage(new File("uploads/images/" . $imgPath));
+            if (count($validator->validate($tmpProductCategory)) != 0) {
+                return new JsonResponse([
+                    'error' => 'Cannot import data from csv. Data not valid.'
+                ], 500);
+            }
+            $tmpProductCategory->setMainImage($imgPath);
+            $tmpProductsCategories[] = $tmpProductCategory;
+        }
+        foreach ($tmpProductsCategories as $productCategory)
+        {
+            try
+            {
+                $em->persist($productCategory);
+                $em->flush();
+            }catch(\Exception $exception)
+            {
+                return new JsonResponse([
+                    'error' => 'Cannot save it in database, please try later'
+                ], 500);
+            }
+
+        }
+
+        return new JsonResponse([
```

Comment:
here we could add number of imported categories for a quick check if everything imported as supposed to

Reply:
Number of imported categories should be in the message so the user could easly confirm success of the import, or it sould say imported all categories when all categories from csv imported successfully